internal/store/store.go

```diff
@@ -21,27 +28,123 @@ func NewStore(ctx context.Context, connString string) (*Store, error) {
 func (db *Store) CreateUsersTable(ctx context.Context) error {
 	_, err := db.Exec(ctx,
 		`CREATE TABLE IF NOT EXISTS users (
-			id bigint GENERATED ALWAYS AS IDENTITY,
+			id bigint GENERATED ALWAYS AS IDENTITY PRIMARY KEY,
 			login text NOT NULL UNIQUE,
-			password text NOT NULL)`)
+			password text NOT NULL,
+			sum double precision)`)
+	if err != nil {
+		return err
+	}
+	_, err = db.Exec(ctx,
+		`CREATE INDEX IF NOT EXISTS users_login_idx ON users (login)`)
 	return err
 }
 
-func (db *Store) AddUser(login string, hash []byte) (int, error) {
-	row := db.QueryRow(context.Background(), "INSERT INTO users (login, password) VALUES ($1, $2) RETURNING id", login, hash)
-	var userID int
-	err := row.Scan(&userID)
+func (db *Store) AddUser(ctx context.Context, u User) (int, error) {
+	row := db.QueryRow(ctx, "INSERT INTO users (login, password) VALUES ($1, $2) RETURNING id", u.Login, u.Hash)
+	err := row.Scan(&u.ID)
 	if err != nil {
 		return 0, err
 	}
-	return userID, nil
+	return u.ID, nil
+}
+
+func (db *Store) GetUser(ctx context.Context, login string) (u User, err error) {
```

Comment:
Думаю можно в общем случае использовать указатель на `*User` для возвращаемого значения.

```suggestion
func (db *Store) GetUser(ctx context.Context, login string) (u *User, err error) {
```